R/npn_get_species.R

```diff
@@ -40,74 +48,107 @@ npn_species_id <- function(ids, ...) {
 #' @export
 #' @rdname npn_species
 #' @param state A US postal state code to filter results.
-#' @param kingdom Filters results by taxonomic kingdom. Takes either 'Animalia' or 'Plantae'.
-#' @return A data frame with information on species in the NPN database whose distribution includes a given state.
+#' @param kingdom Filters results by taxonomic kingdom. Valid values include
+#'   `'Animalia'`, `'Plantae'`, or `NULL` (the default, which returns results
+#'   for both).
+#' @param ... Currently unused.
+#' @return A tibble with information on species in the NPN database whose
+#'   distribution includes a given state.
 #' @examples \dontrun{
 #' npn_species_state(state = "AZ")
 #' npn_species_state(state = "AZ", kingdom = "Plantae")
 #' }
 npn_species_state <- function(state, kingdom = NULL, ...) {
-  args <- npnc(list(state = state, kingdom = kingdom))
-  ldfply(npn_GET(paste0(base(), 'species/getSpeciesByState.json'), args, ...))
+  #TODO: clarify if multiple states are accepted or not.
+  # The state[1]=CA&state[2]=AZ form appears to work with the API
```

Comment:
This appears to work, but for now at least we are limiting to one state so it is clear which results belong to which state